src/msca/metrics/main.py

```diff
@@ -0,0 +1,475 @@
+from typing import Optional, List, Union
+from enum import StrEnum
+import numpy as np
+import pandas as pd
+
+from sklearn import metrics
+
+class ErrorMetric(StrEnum):
+    """
+    Examples
+    --------
+    >>> # Simple usage
+    >>> metric = ErrorMetric.MEAN_ABSOLUTE_ERROR
+    >>> score = metric.eval(df, "obs", "pred", "weights")
+    >>> 
+    >>> # Grouped calculation
+    >>> grouped_scores = metric.eval(df, "obs", "pred", "weights", groupby=["region"])
+    """
+    MEAN_ABSOLUTE_ERROR = "mean_absolute_error"
+    MEAN_ABSOLUTE_PERCENTAGE_ERROR = "mean_absolute_percentage_error"
+    MEAN_SQUARED_ERROR = "mean_squared_error"
+    MEDIAN_ABSOLUTE_ERROR = "median_absolute_error"
+    ROOT_MEAN_SQUARED_ERROR = "root_mean_squared_error"
+    
+    def eval(
+        self, 
+        data: pd.DataFrame, 
+        obs: str, 
+        pred: str, 
+        weights: str,
+        groupby: Optional[List[str]] = None
+    ) -> Union[float, pd.DataFrame]:
+        """
+        Evaluate the error metric on the provided data.
+        
+        Parameters
+        ----------
+        data : pd.DataFrame
+            Input DataFrame containing all required columns
+        obs : str
+            Column name for observed/actual values
+        pred : str
+            Column name for predicted values
+        weights : str
+            Column name for sample weights
+        groupby : List[str], optional
+            Column names to group by for grouped calculations
+            
+        Returns
+        -------
+        Union[float, pd.DataFrame]
+            Single metric value if no groupby, DataFrame with grouped results if groupby specified
+        """
+        if groupby is not None:
+            return self._eval_grouped(data, obs, pred, weights, groupby)
+        
+        # Extract arrays from DataFrame columns
+        obs_values = data[obs].values
```

Comment:
Let's use `.to_numpy()` rather than `.values`
https://docs.astral.sh/ruff/rules/pandas-use-of-dot-values/